include/internal/rcu.h

```diff
@@ -0,0 +1,46 @@
+/*
+ *
+ * Copyright 2023 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+
+
+#ifndef OPENSSL_RCU_H
+# define OPENSSL_RCU_H
+# pragma once
+
+struct rcu_qp;
+
+typedef void (*rcu_cb_fn)(void *data);
+
+struct rcu_cb_item {
+    rcu_cb_fn fn;
+    void *data;
+    struct rcu_cb_item *next;
+};
+
+struct crypto_rcu_lock_st {
+    struct rcu_qp *current_qp;
+};
```

Comment:
Why do we have this public structure?

Reply:
We did need it some time ago, but so many changes have been made to privatize the internals, that I guess it could be eliminated and the public facing rcu lock be typedefed to void*